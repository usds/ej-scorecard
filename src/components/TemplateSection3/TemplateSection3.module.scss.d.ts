declare namespace TemplateSection3Namespace {
  export interface ITemplateSection3Scss {
    templateSection3Container: string;
  }
}

declare const TemplateSection3ScssModule: TemplateSection3Namespace.ITemplateSection3Scss & {
  /** WARNING: Only available when "css-loader" is used without "style-loader" or "mini-css-extract-plugin" */
  locals: TemplateSection3Namespace.ITemplateSection3Scss;
};

export = TemplateSection3ScssModule;
