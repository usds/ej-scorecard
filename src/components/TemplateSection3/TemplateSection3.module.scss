@use "uswds-core" as *;

.templateSection3Container{

}
