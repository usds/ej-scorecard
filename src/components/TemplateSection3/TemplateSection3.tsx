import React from 'react';
import {
  Grid,
  SummaryBox,
  SummaryBoxContent,
  SummaryBoxHeading,
} from '@trussworks/react-uswds';
// import * as styles from './TemplateSection3.module.scss';

import { TemplateSectionProps } from '@/types';
import { DELIMITERS } from '@/data/constants';
import { possessivePlural, splitTextIntoParagraphs } from '../util';

const TemplateSection3: React.FC<TemplateSectionProps> = ({ agencyData }) => {
  return (
    <>
      <Grid row>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <h2 className={`scorecard-header2`}>
            Institutionalizing Environmental Justice
          </h2>
        </Grid>
      </Grid>
      <Grid row gap={6}>
        <Grid desktop={{ col: 8 }} tablet={{ col: 10 }} col={12}>
          <p className={`margin-top-0`}>
            Since the start of the Biden-Harris Administration, federal agencies
            have been working to embed environmental justice into all aspects of
            their work, including by implementing
            {` `}
            <a
              className="usa-link usa-link--alt usa-link--external"
              href={`https://www.archives.gov/files/federal-register/executive-orders/pdf/12898.pdf`}
              target="_blank"
              rel="noreferrer"
            >
              Executive Order 12898
            </a>
            {` `}
            and
            {` `}
            <a
              className="usa-link usa-link--alt usa-link--external"
              href={`https://www.federalregister.gov/documents/2021/02/01/2021-02177/tackling-the-climate-crisis-at-home-and-abroad#:~:text=Sec.%20223.%20Justice40,40-percent%20goal.`}
              target="_blank"
              rel="noreferrer"
            >
              Executive Order 14008
            </a>
            . The following information shares some of the work{` `}
            <strong>{agencyData.THE_A_NAME}</strong> has done from January 2021
            to September 2022 to institutionalize environmental justice.
          </p>

          <p>Phase One Scorecard metrics and highlights include:</p>

          <ul className={`bullet1`}>
            {agencyData.PLAN_BOOL && agencyData.PLAN_UPDATE && (
              <li>
                <strong>{agencyData.PLAN_BOOL}</strong> an environmental justice
                Strategic Plan that <strong>{agencyData.PLAN_UPDATE}</strong>
                {` `}
                updated in the past 5 years
              </li>
            )}
            <ul>
              {agencyData.EJP_E1 && <li>{agencyData.EJP_E1}</li>}
              {agencyData.EJP_E2 && <li>{agencyData.EJP_E2}</li>}
              {agencyData.EJP_E3 && <li>{agencyData.EJP_E3}</li>}
            </ul>
            {agencyData.EJTOOL_N && (
              <li>
                <strong>{agencyData.EJTOOL_N}</strong> environmental justice
                tool(s) or resource(s)
              </li>
            )}
            {agencyData.EJSTAFF_N && (
              <li>
                <strong>{agencyData.EJSTAFF_N}</strong> staff that work on
                environmental justice, either in a full- or part-time capacity
              </li>
            )}
            {agencyData.EJTRAIN_N && (
              <li>
                <strong>{agencyData.EJTRAIN_N}</strong> trainings for staff on
                environmental justice
              </li>
            )}
            {agencyData.EJ_OFFICE && (
              <li>
                <strong>{agencyData.EJ_OFFICE}</strong> new or strengthened
                office(s) on environmental justice
              </li>
            )}
            {agencyData.EJWG_N && (
              <li>
                <strong>{agencyData.EJWG_N}</strong> new or strengthened office
                internal working group(s)/steering committee(s)/council(s) on
                environmental justice{` `}
              </li>
            )}
            {agencyData.EJP_NAR && <li>{agencyData.EJP_NAR}</li>}
          </ul>
        </Grid>
        <Grid desktop={{ col: 4 }} tablet={{ col: 10 }} col={12}>
          {agencyData.IEJ_HIGH && (
            <SummaryBox>
              <SummaryBoxHeading headingLevel="h3">
                {`${possessivePlural(
                  agencyData.A_ACRONYM,
                )} Institutionalizing Environmental Justice highlights`}
              </SummaryBoxHeading>
              <SummaryBoxContent>
                {agencyData.IEJ_HIGH.includes(DELIMITERS.PARAGRAPH)
                  ? splitTextIntoParagraphs(
                      agencyData.IEJ_HIGH,
                      DELIMITERS.PARAGRAPH,
                    ).map((paragraph) => paragraph)
                  : agencyData.IEJ_HIGH}
              </SummaryBoxContent>
            </SummaryBox>
          )}
        </Grid>
      </Grid>
    </>
  );
};

export default TemplateSection3;
