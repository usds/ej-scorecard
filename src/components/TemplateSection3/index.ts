import TemplateSection3 from './TemplateSection3';
export default TemplateSection3;
