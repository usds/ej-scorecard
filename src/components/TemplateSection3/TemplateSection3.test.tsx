import createComponentWithIntl from '@/test/testHelpers';

import TemplateSection3 from './TemplateSection3';

const agencyData = {
  id: `39455749-3310-53aa-b5fa-22590adc1e05`,
  A_NAME: `Environmental Protection Agency`,
  A_LOGO: `epa.png`,
  A_ADD1: `1200 Pennsylvania Avenue N.W.`,
  A_ADD2: `Washington, D.C. 20004`,
  A_PHONE: `(202) 564-4700`,
  A_URL: `epa.gov`,
  A_MISSION: `The Environmental Protection Agency (EPA) protects human health and the environment. EPA actively seeks to incorporate Environmental Justice at all levels of the agency, along with strategic partnerships with other federal, state, local, and tribal departments.`,
  J40_URLTXT: `epa.gov/justice40`,
  J40_URLLINK: `epa.gov/justice40`,
  A_EMAIL: ``,
  J40_TCP: `40`,
  J40_ANNOUNCE: `80`,
  J40_AMOUNT: `$28 million`,
  J40_E1: `Developed resources for states revising their disadvantaged community definitions and affordability criteria through the Clean Water State Revolving Fund (CWSRF), the Drinking Water State Revolving Fund (DWSRF), the Water Infrastructure Improvements for the Nation (WIIN), and Bipartisan Infrastructure Law (BIL) funding`,
  J40_E2: `Engaged with State Revolving Fund partners to prioritize and launch technical assistance pilots for disadvantaged communities`,
  J40_E3: `Prioritized financing projects in economically stressed communities in the Water Infrastructure Finance and Innovation Act (WIFIA) program FY22 Notice of Funding Availability (NOFA)`,
  J40_MOD1: ``,
  J40_MOD2: ``,
  J40_MOD3: ``,
  J40_HIGH: `•\tProvided assistance to [XX] communities through the Technical Assistance Services for Communities (TASC) program  •\tWorked with over 200 Tribal partners to design, develop, and publish curriculum about how to reduce childhood lead exposure, and hosted 10 training sessions in Tribal communities, reaching 450 participants  •\tImproved multilingual access to information, including Requests for Proposals, webinars, and in-person trainings  •\tConducted and participated in 27 federal outreach and engagement events in support of small and disadvantaged businesses, reaching over 700 participants in FY22`,
  A_STRATPLANTXT: ``,
  A_STRATPLANLINK: ``,
  A_NEPA: `EPA outlines NEPA policies and guidelines on their website, including Executive Order 12898, key steps in the NEPA process, how to approach public participation in decision making, and EJScreen—an Environmental Justice mapping and screening tool`,
  NUM_NEPA: `1`,
  NEPA_H1: `Hosted 3 public meetings with 200 participants regarding projects to reduce Tijuana River watershed pollution that affects communities in Tijuana/San Diego`,
  NEPA_H2: `Additional NEPA highlight`,
  NEPA_H3: `Additional NEPA highlight`,
  NEPA_TRAIN: `421`,
  TITLEVI_NAR: `Most complaints under investigation involve state and local agencies and allege that actions or inactions have disparate impacts, and have subjected people to discrimination on the basis of race, national origin, disability, and/or age. Complaints involve many types of facilities and services, including: petrochemical, asphalt, concrete, recycling, CAFO, and water treatment and management services/facilities.`,
  TITLEVI_N: `28`,
  TITLEVI_R_N: `2`,
  A_NAR: `Formal partnerships between EPA and non-federal entities helps to advance Environmental Justice and/or inform policy and decision-making. Additionally, cooperative agreements support formal partnership agreements with states, Tribes, and community-based organizations.`,
  TA_N: `130`,
  TA_E1: `Events/webinars/workshops through The Environmental Justice Thriving Communities Technical Assistance Centers`,
  TA_E2: `Closing America’s Wastewater Access Gap`,
  TA_E3: `Technical assistance calls with communities with regional offices`,
  PP_N: `480`,
  PP_E1: `2 Journey to Justice tours, where EPA Administrator Michael S. Regan heard firsthand from communities facing Environmental Justice concerns in Puerto Rico, Mississippi, Louisiana, and Texas`,
  PP_E2: `5 public engagement sessions, attended by over 500 people, at 276 Superfund sites where there are environmental concerns. Participants had the opportunity to comment on the Office of Land and Emergency Management Environmental Justice Action Plan.`,
  PP_E3: `Office of Environmental Justice and External Civil Rights hosted 19 National Environmental Justice Community Engagement calls to inform the public about EPA's Environmental Justice work and enhance opportunities to maintain an open dialogue with advocates.`,
  TC_NAR: `EPA issued the Policy on Environmental Justice for Working with Federally Recognized Tribes and Indigenous Peoples in 2014, to advance Environmental Justice for the protection of the environment and public health`,
  TC_N: `18`,
  TC_E1: `Formal consultation and coordination with Indian Tribal Governments, as well as public engagement events with communities, on risk management for nine of the first 10 chemicals under risk evaluations by EPA`,
  TC_E2: `“EJ, Tribes, and Indigenous Peoples” training webinar in May 2022`,
  TC_E3: `“Reflections on What Indigenous Research Means,” with Dr. Kyle Whyte at the May 2021 Tribal Science Council (TSC) meeting`,
  ECR_HIGH: `Environmental and Civil Rights highlight space`,
  PLAN_BOOL: `Has`,
  PLAN_UPDATE: `was not`,
  EJP_E1: ``,
  EJP_E2: ``,
  EJP_E3: ``,
  EJTOOL_N: `70`,
  EJSTAFF_N: `206`,
  EJTRAIN_N: `25`,
  EJWG_N: `20`,
  EJP_NAR: ``,
  IEJ_HIGH: `Institutionalizing Environmental Justice    Federal agencies have been working to institutionalize Environmental Justice in keeping with their unique missions and the statutory authorities that govern their work, including to continue to implement Executive Order 12898, and to advance the objectives of Executive Order 14008. The following section describes example activities that the agency has taken from January 2021 to September 2022.    •\thas an Environmental Justice related Strategic Plan that was not updated in the last five years  •\t25 policies developed or updated related to Environmental Justice   •\t70 Environmental Justice tools or resources  •\t206 staff, either in a full- or part-time capacity, working on Environmental Justice  •\t421 trainings for staff on Environmental Justice, as a general study or regarding civil rights specifically  •\t20 Environmental Justice working groups, steering committees and/or councils    Additional Institutionalizing Environmental Justice Highlights at EPA    o\tAdded Environmental Justice components to different EPA efforts, such as the Border 2025 program guiding principles, adding “inclusion” as a cross-cutting research priority, and adding Environmental Justice as a component in the Excellence in Site Reuse Awards nomination form   o\tStrategic Research Action Plans (StRAPs) included approximately 145 research products that are "Environmental Justice-focused" or "Environmental Justice-relevant”  o\tIncluded Environmental Justice considerations into numerous competitive notice of funding opportunities.`,
};
describe(`rendering of TemplateSection3 Component`, () => {
  it(`checks if component renders`, () => {
    const component = createComponentWithIntl(
      <TemplateSection3 agencyData={agencyData} />,
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
