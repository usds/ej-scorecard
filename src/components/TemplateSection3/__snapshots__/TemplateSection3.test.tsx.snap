// Vitest Snapshot v1

exports[`rendering of AgencyInfo Component > checks if component renders group 1 1`] = `
<div
  className="grid-row grid-gap"
  data-testid="grid"
>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
    data-testid="grid"
  >
    <div
      className="grid-row grid-gap"
      data-testid="grid"
    >
      <div
        className="grid-col-4"
        data-testid="grid"
      >
        <div>
          <div
            className="gatsby-image-wrapper gatsby-image-wrapper-constrained"
            data-gatsby-image-wrapper=""
            style={
              {
                "backgroundColor": undefined,
                "display": "inline-block",
                "overflow": "hidden",
                "position": "relative",
                "verticalAlign": "top",
              }
            }
          >
            <div
              style={
                {
                  "display": "block",
                  "maxWidth": 334,
                }
              }
            >
              <img
                alt=""
                aria-hidden="true"
                role="presentation"
                src="data:image/svg+xml;charset=utf-8,%3Csvg%20height='278'%20width='334'%20xmlns='http://www.w3.org/2000/svg'%20version='1.1'%3E%3C/svg%3E"
                style={
                  {
                    "display": "block",
                    "maxWidth": "100%",
                    "position": "static",
                  }
                }
              />
            </div>
            <div
              aria-hidden={true}
              data-placeholder-image=""
              style={
                {
                  "height": "100%",
                  "left": 0,
                  "position": "absolute",
                  "top": 0,
                  "width": "100%",
                }
              }
            />
            <picture>
              <source
                data-srcset="/static/703cdc54fb8f1bddfe59854183c3829b/59647/agency-a.webp 84w,/static/703cdc54fb8f1bddfe59854183c3829b/02be2/agency-a.webp 167w,/static/703cdc54fb8f1bddfe59854183c3829b/7adb0/agency-a.webp 334w"
                sizes="(min-width: 334px) 334px, 100vw"
                type="image/webp"
              />
              <img
                alt="Agency logo"
                data-gatsby-image-ssr=""
                data-main-image=""
                data-src="/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png"
                data-srcset="/static/703cdc54fb8f1bddfe59854183c3829b/c3bfd/agency-a.png 84w,/static/703cdc54fb8f1bddfe59854183c3829b/53761/agency-a.png 167w,/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png 334w"
                decoding="async"
                loading="lazy"
                sizes="(min-width: 334px) 334px, 100vw"
                style={
                  {
                    "backgroundColor": undefined,
                    "height": "100%",
                    "left": 0,
                    "objectFit": undefined,
                    "objectPosition": undefined,
                    "opacity": 0,
                    "position": "absolute",
                    "top": 0,
                    "transform": "translateZ(0)",
                    "transition": "opacity 250ms linear",
                    "width": "100%",
                    "willChange": "opacity",
                  }
                }
              />
            </picture>
            <noscript>
              <picture>
                <source
                  sizes="(min-width: 334px) 334px, 100vw"
                  srcSet="/static/703cdc54fb8f1bddfe59854183c3829b/59647/agency-a.webp 84w,/static/703cdc54fb8f1bddfe59854183c3829b/02be2/agency-a.webp 167w,/static/703cdc54fb8f1bddfe59854183c3829b/7adb0/agency-a.webp 334w"
                  type="image/webp"
                />
                <img
                  alt="Agency logo"
                  data-gatsby-image-ssr=""
                  data-main-image=""
                  decoding="async"
                  loading="lazy"
                  sizes="(min-width: 334px) 334px, 100vw"
                  src="/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png"
                  srcSet="/static/703cdc54fb8f1bddfe59854183c3829b/c3bfd/agency-a.png 84w,/static/703cdc54fb8f1bddfe59854183c3829b/53761/agency-a.png 167w,/static/703cdc54fb8f1bddfe59854183c3829b/dff88/agency-a.png 334w"
                  style={
                    {
                      "backgroundColor": undefined,
                      "height": "100%",
                      "left": 0,
                      "objectFit": undefined,
                      "objectPosition": undefined,
                      "opacity": 0,
                      "position": "absolute",
                      "top": 0,
                      "transform": "translateZ(0)",
                      "transition": "opacity 250ms linear",
                      "width": "100%",
                      "willChange": "opacity",
                    }
                  }
                />
              </picture>
            </noscript>
            <script
              dangerouslySetInnerHTML={
                {
                  "__html": "const t=\\"undefined\\"!=typeof HTMLImageElement&&\\"loading\\"in HTMLImageElement.prototype;if(t){const t=document.querySelectorAll(\\"img[data-main-image]\\");for(let e of t){e.dataset.src&&(e.setAttribute(\\"src\\",e.dataset.src),e.removeAttribute(\\"data-src\\")),e.dataset.srcset&&(e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\"));const t=e.parentNode.querySelectorAll(\\"source[data-srcset]\\");for(let e of t)e.setAttribute(\\"srcset\\",e.dataset.srcset),e.removeAttribute(\\"data-srcset\\");e.complete&&(e.style.opacity=1,e.parentNode.parentNode.querySelector(\\"[data-placeholder-image]\\").style.opacity=0)}}",
                }
              }
              type="module"
            />
          </div>
          <h3>
            Test Agency
          </h3>
          <span>
            1 Pennsylvania Avenue N.W.
          </span>
          <span>
            Washington, D.C. 20004
          </span>
          <span>
            (212) 533-3333
          </span>
          <span>
            <a
              className="usa-link usa-link--alt usa-link--external"
              href="https://someagency.gov"
              rel="noopener noreferrer"
              target="_blank"
            >
              someagency.gov
            </a>
          </span>
        </div>
      </div>
      <div
        className="grid-col-8"
        data-testid="grid"
      >
        <div>
          <h2>
            About the Test Agency
          </h2>
          <p>
            mission
          </p>
           
        </div>
      </div>
    </div>
    <div
      className="grid-row grid-gap"
      data-testid="grid"
    >
      <p>
        This page provides information on how
         
        <strong>
          the Test Agency
        </strong>
         is working to reduce climate and environmental burdens and harms in communities across America. It is part of Phase One of the Biden-Harris Administration's Environmental Justice Scorecard.
      </p>
    </div>
  </div>
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-1"
    data-testid="grid"
  />
  <div
    className="grid-col-12 tablet:grid-col-10 desktop:grid-col-3"
    data-testid="grid"
  />
</div>
`;

exports[`rendering of ScorecardSideNav Component > checks if component renders 1`] = `null`;

exports[`rendering of TemplateSection3 Component > checks if component renders 1`] = `
[
  <div
    className="grid-row"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <h2
        className="scorecard-header2"
      >
        Institutionalizing Environmental Justice
      </h2>
    </div>
  </div>,
  <div
    className="grid-row grid-gap-6"
    data-testid="grid"
  >
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-8"
      data-testid="grid"
    >
      <p
        className="margin-top-0"
      >
        Since the start of the Biden-Harris Administration, federal agencies have been working to embed environmental justice into all aspects of their work, including by implementing
         
        <a
          className="usa-link usa-link--alt usa-link--external"
          href="https://www.archives.gov/files/federal-register/executive-orders/pdf/12898.pdf"
          rel="noreferrer"
          target="_blank"
        >
          Executive Order 12898
        </a>
         
        and
        <a
          className="usa-link usa-link--alt usa-link--external"
          href="https://www.federalregister.gov/documents/2021/02/01/2021-02177/tackling-the-climate-crisis-at-home-and-abroad#:~:text=Sec.%20223.%20Justice40,40-percent%20goal."
          rel="noreferrer"
          target="_blank"
        >
          Executive Order 14008
        </a>
        . The following information shares some of the work
         
        <strong>
          the Test Agency
        </strong>
         has done from January 2021 to September 2022 to institutionalize environmental justice.
      </p>
      <p>
        <strong>
          Phase One Scorecard metrics and highlights include:
        </strong>
      </p>
      <ul
        className="bullet1"
      >
        <li>
          <strong>
            Has
          </strong>
           an environmental justice Strategic Plan that 
          <strong>
            was not
          </strong>
           
          updated in the past 5 years
        </li>
        <ul />
        <li>
          <strong>
            100
          </strong>
           environmental justice tool(s) or resource(s)
        </li>
        <li>
          <strong>
            100
          </strong>
           staff that work on environmental justice, either in a full- or part-time capacity
        </li>
        <li>
          <strong>
            100
          </strong>
           trainings for staff on environmental justice, as a general environmental justice training or Civil Rights and environmental justice training
        </li>
        <li>
          <strong>
            100
          </strong>
           new or strengthened office on environmental justice
        </li>
        <li>
          <strong>
            100
          </strong>
           new or strengthened office internal working group(s)/steering committee(s)/council(s) on environmental justice
           
        </li>
      </ul>
    </div>
    <div
      className="grid-col-12 tablet:grid-col-10 desktop:grid-col-4"
      data-testid="grid"
    >
      <div
        className="usa-summary-box"
        data-testid="summary-box"
      >
        <div
          className="usa-summary-box__body"
        >
          <h3
            className="usa-summary-box__heading"
          >
            TAG's Institutionalizing Environmental Justice highlights
          </h3>
          <div
            className="usa-summary-box__text"
          >
            test
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
